src/components/popup-dialog/PopupDialog.tsx

```diff
@@ -22,33 +27,88 @@ const PopupDialog: FC<PopupDialogProps> = ({
   closeModalAfterDelay
 }) => {
   const { isMobile } = useBreakpoints()
+  const [showConfirm, setShowConfirm] = useState(false)
 
   const handleMouseOver = () => timerId && clearTimeout(timerId)
   const handleMouseLeave = () => timerId && closeModalAfterDelay()
-  const handleClose = () => closeModalAfterDelay(0)
+  const handleClose = () => setShowConfirm(true)
+
+  const confirmCloseHandler = () => {
+    setShowConfirm(false)
+    closeModalAfterDelay(0)
+  }
+  const cancelCloseHandler = () => setShowConfirm(false)
 
   return (
-    <Dialog
-      PaperProps={paperProps}
-      data-testid='popup'
-      disableRestoreFocus
-      fullScreen={isMobile}
-      maxWidth='xl'
-      onClose={handleClose}
-      open
-    >
-      <Box
-        data-testid='popupContent'
-        onMouseLeave={handleMouseLeave}
-        onMouseOver={handleMouseOver}
-        sx={styles.box}
+    <>
+      <Dialog
+        PaperProps={paperProps}
+        data-testid='popup'
+        disableRestoreFocus
+        fullScreen={isMobile}
+        maxWidth='xl'
+        onClose={handleClose}
+        open
+      >
+        <Box
+          data-testid='popupContent'
+          onMouseLeave={handleMouseLeave}
+          onMouseOver={handleMouseOver}
+          sx={styles.box}
+        >
+          <IconButton onClick={handleClose} sx={styles.icon}>
+            <CloseIcon />
+          </IconButton>
+          <Box sx={styles.contentWraper}>{content}</Box>
+        </Box>
+      </Dialog>
+      <Dialog
```

Comment:
Why do we render two dialogs in this component?

Reply:
Fixed it, and made the confirmation window conditional rendering